_posts/2019-05-21-i18n-messages.adoc

```diff
@@ -0,0 +1,327 @@
+
+1. Community Languages Packs
+2. Manual providing Translations
+
+
+=== Community Language Packs
+
+The first and easiest solution is to use existing language packs that are created by the community. There are currently twelve Translations available including Spanisch, Russian, French, Chinese, German and a lot more.
+
+WARNING: The Community Translations Packs are provided by the community, therefore the accuracy and the catch up with the latest changes in the platform differ and are provided as best effort.
+
+In order to get those translations into the Application they can be fetched manually from the https://github.com/cuba-platform/translations[CUBA Translations Github repository]. Some Translations are also directly available via an application component through the https://www.cuba-platform.com/marketplace/[CUBA marketplace],
+which makes the installation as easy as installing any other application component.
+
+Afterwards, the only thing to do is to add the new language to the application property `cuba.availableLocales`.
+
+=== Manual Providing Translations
+
+In case the target language is not available through the community or some translations are missing or are not accurate, it is also possible to provide those translations directly within the application.
+
+CUBA itself provides translations for the screens directly in the source code of the platform as regular `*.properties` files. The User browse screen for example contains the https://github.com/cuba-platform/cuba/blob/master/modules/gui/src/com/haulmont/cuba/gui/app/security/user/browse/messages.properties[messages.properties] file in the package `com.haulmont.cuba.gui.app.security.user.browse`,
+which contains the translations for the screen.
+
+In order to provide / override translations for this screen, two things have to be done:
+
+==== Create GUI module if Necessary
+
+The destination module of where the translations have to be placed has to be available in the application. The majority of the platform screens are placed in the GUI module, which is not available for 7.0+ CUBA applications by default. In case the target application does not have the GUI module it has to be included into the application to place the translation files into it.
+
+TIP: Adding the GUI module via CUBA Studio can be done via the Menu: `CUBA > Advanced > Manage Modules > Create 'gui' module`.
+
+
+==== Create Package-Mirror in the Application
+
+After ensuring the target module is available in the application, the package where the original translation files exist, have to be created in the application as well. In this directory the additional / replacing message files are placed. CUBA will pick up the files and display the correct translations for the screen.
+
+In the Petclinic example, the destination target package has been created for the Users browse screen:
```

Comment:
If you include the whole German translation to the app, you will have to choose another language for this example...